java-checks/src/main/java/org/sonar/java/symexec/ExecutionState.java

```diff
@@ -193,35 +147,50 @@ ExecutionState setBooleanConstraint(SymbolicValue.SymbolicVariableValue variable
     return this;
   }
 
-  void invalidateRelationsOnValue(SymbolicValue value) {
-    Multimap<SymbolicValue, SymbolicValue> pairs = HashMultimap.create();
-    for (ExecutionState current = this; current != null; current = current.parentState) {
-      pairs.putAll(value, current.findRelatedValues(value));
+  void invalidateFields() {
+    Set<Symbol> assignedVariables = new HashSet<>();
+    for (ExecutionState state = this; state != null; state = state.parentState) {
+      assignedVariables.addAll(state.values.keySet());
     }
-    for (Map.Entry<SymbolicValue, SymbolicValue> entry : pairs.entries()) {
-      setRelation(entry.getKey(), SymbolicRelation.UNKNOWN, entry.getValue());
+    for (Symbol variable : assignedVariables) {
+      if (isField(variable)) {
+        assignValue(variable, new SymbolicValue());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this instantiation of "SymbolicValue". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2440)